### PR TITLE
CI-33 Add a JDK11 job

### PR DESCRIPTION
**Pull request by MarkWareham**
# Description
This is a CI addition. 

Adds a new Jenkins pipeline for testing PRs against JDK11

# Testing

### Testing Performed

It is not possible to test this without placing available in master for the Jenkins server to access.
Change is low risk. A copy of existing job with JDK versions changed.
Testing will be performed after this PR.

# Documentation
Internal wiki updated.

# Notes for Reviewers
Job is a copy of the existing job, except the tools block is moved to each stage as we want to build with JDK8 and test with JDK11

### CODE REVIEW

**Comment by MarkWareham** [comment]
jenkins test jdk-11 please

**Comment by MarkWareham** [comment]
Jenkins test JDK-11 please

**Comment by MarkWareham** [comment]
test failed as the jenkinsfile not available. Must need testing after merge

**Comment by Pandrex247** [comment]
@MarkWareham Are you sure?
`Jenkinsfile-jdk11` vs. `Jenkinsfile-JDK11`

**Comment by Pandrex247** [comment]
Jenkins test JDK-11 please

**Comment by Pandrex247** [comment]
Fixed it on Jenkins side, build is now progressing, but it seems to still be using JDK8...
`Java version: 1.8.0_212, vendor: Azul Systems, Inc., runtime: /usr/lib/jvm/zulu-8-amd64/jre`

**Comment by MarkWareham** [comment]
@Pandrex247 I realised it was the caps. See my two sequential comments. It was a mistake in the regex for the trigger phrase

**Comment by Pandrex247** [comment]
> 
> 
> @Pandrex247 I realised it was the caps. See my two sequential comments. It was a mistake in the regex for the trigger phrase

I meant caps in the Jenkins job itself, not the trigger phrase - Jenkins was looking for `Jenkinsfile-jdk11` but the file is `Jenkinsfile-JDK11`

File: Jenkinsfile-JDK11 [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                    archiveArtifacts artifacts: 'appserver/extras/payara-micro/payara-micro-distribution/target/payara-micro.jar', fingerprint: true
+                }
+            }
+        }
+        stage('Setup for Quicklook Tests') {
+            tools {
+                jdk "zulu-11"
+            }
+            steps {
+                sh "rm *.zip"
```
Comment (Pandrex247) [inline Jenkinsfile-JDK11]: Build fails here

Comment (Pandrex247) [reply Jenkinsfile-JDK11]: This seems to be trying to delete the log files from previous runs: https://github.com/payara/Payara/pull/4315/files#diff-28788dd8800bef08d2db40a4380a2f1eR71

The following change might work, haven't tried it within a Jenkinsfile (and not sure on the placement of the quotes, pick which you think is right):
```suggestion
                sh "rm *.zip || true"
```
```suggestion
                sh "rm *.zip" || true
```

**Comment by MarkWareham** [comment]
Jenkins test JDK-11 please

**Pull request closed without merging**